src/Form.jsx

```diff
@@ -20,14 +20,29 @@ class Form extends React.Component {
 
     this.progressFields = [];
     Object.keys(fields).forEach(name => {
-      initial[name] = null;
-      if (fields[name].type === "checkboxGroup") {
-        initial[name] = [];
+      let value = fields[name].defaultValue;
+
+      if (typeof value === 'undefined') {
+        value = null;
+      }
+
+      // this is a controlled field. let's ignore defaultValue
```

Comment:
This code should not be necessary. We should not be checking whether someone was silly enough to set a default value on a controlled field.